File: ESDEjercicios/ListasEnlazadas/ListaEnlazadaPOO/nodo_lista.cpp
#include <iostream>
using namespace std;

class nodo_lista
{
    int dato;
    nodo_lista *sig;

public:
    nodo_lista(int d);
    void inserta_s(nodo_lista *nodo);
    void muestra();
};
nodo_lista::nodo_lista(int d)
{
    dato = d;
    sig = NULL;
}
void nodo_lista::inserta_s(nodo_lista *nodo)
{
    if (sig == NULL) // 0
    {
        sig = nodo;
        nodo->sig = NULL; // 0
    }
    /*else {
    nodo->sig = sig;
    sig = nodo;
    }*/
}
void nodo_lista::muestra()
{
    nodo_lista *p1;
    p1 = this;
    while (p1 != 0)
    {
        cout << p1->dato << " ";
        p1 = p1->sig;
    }
    cout << '\n';
}

File: ESDEjercicios/AppListaEnlazada/src/NodoLista.java
public class NodoLista {
    int dato;
    NodoLista sig;

    NodoLista(int d){
        dato = d;
        sig = null;
    }

    void insertarNodo(NodoLista nodo){
        if (sig==null) {
            this.sig = nodo;
            nodo.sig = null;
        }
    }

    void muestra(){
        NodoLista p1;
        p1 = this;

        while(p1 != null){
            System.out.print("| "+p1.dato+" -> ");
            p1 = p1.sig;
        }
        System.out.println("null");
        System.out.println();
    }
}
